components/Map/hooks/useMapActions.tsx

```diff
@@ -0,0 +1,93 @@
+import { useUser } from '@auth0/nextjs-auth0';
+import { useContext, useRef } from 'react';
+import { InterestZone } from '../../../models/zone.model';
+import useInterestZones from '../../hooks/useInterestZones';
+import { ModalContext } from '../../Providers/ModalProvider';
+import { InterestZoneProviderContext } from '../../Providers/ZoneProvider';
+
+interface Bounds {
+  north: number;
+  south: number;
+  east: number;
+  west: number;
+}
+const CLUJ_NAPOCA_BOUNDS: Bounds = {
+  north: parseFloat(process.env.NEXT_PUBLIC_NORTH!),
+  south: parseFloat(process.env.NEXT_PUBLIC_SOUTH!),
+  west: parseFloat(process.env.NEXT_PUBLIC_WEST!),
+  east: parseFloat(process.env.NEXT_PUBLIC_EAST!),
+};
+
+export default function useMapActions() {
+  const { user } = useUser();
+  const { setInterestZone, setPartialInterestZone } = useContext(InterestZoneProviderContext);
+  const { setModal } = useContext(ModalContext);
+  const { interestZones } = useInterestZones();
+  const mapRef = useRef<google.maps.Map>();
+
+  const searchOnMap = (address: string): void => {
+    const map = mapRef.current;
+    if (!map) return;
+
+    const geocoder = new google.maps.Geocoder();
+    const markerSearch = new google.maps.Marker();
+
+    address = `${address}, ${process.env.NEXT_PUBLIC_CITY_LOCATION}`;
```

Comment:
Please avoid assigning new values to input arguments. In this particular case, consider renaming the method argument to `partialAddress` and creating a dedicated `address` variable for the above value.